src/routes/snippets.rs

```diff
@@ -57,20 +56,27 @@ impl NewSnippet {
 
 #[post("/snippets", data = "<body>")]
 pub fn create_snippet(
-    origin: &Origin,
     config: State<Config>,
     storage: State<Box<dyn Storage>>,
+    origin: &Origin,
+    requested_content_type: Result<NegotiatedContentType, ApiError>,
     body: Result<Input<NewSnippet>, ApiError>,
-) -> Result<Created<JsonValue>, ApiError> {
+) -> Result<Created<Output<Snippet>>, ApiError> {
+    let NegotiatedContentType(content_type) = requested_content_type?;
```

Comment:
Yeah, this is controversial one :( I would also like to have it work like that. However, we have the following problem.

 `Output` is a `Responder`. The logic of a `Responder` is executed *after* the request handler has given the execution control back to Rocket. It's perfectly fine to perform content negotiation and fail the request at this point, *but* this means that we potentially fail after we've already processed the request and, for example, modified the DB state by creating or updating a snippet, which looks weird to me. I would rather perform the negotiation up front and fail the request right away. Which is what I tried to code here.

You would think there should be a mechanism akin to `before_request()` in Flask that would allow you to *implicitly* execute content negotiation before each request, and there is one - https://api.rocket.rs/v0.4/rocket/fairing/trait.Fairing.html - but it's not possible to abort a request from within a `on_request` fairing: `It may not, however, abort or respond directly to the request; these issues are better handled via request guards or via response callbacks.`

So I think we have 3 options here:

* put this logic into the `Output` responder and do not worry about the potentially failing requests that modify the DB state. This way, this will be completely transparent to the API routes
* make the content negotiation very explicit by requiring the content type to be passed to `Output` as the first argument (which is what  I did here)
* make the content negotiation "a bit less explicit" by requiring the API routes to accept an *unused* `NegotiatedContentType` argument